lms/djangoapps/mobile_api/middleware.py

```diff
@@ -0,0 +1,96 @@
+        request timestamp <= update required timespan for next supported version
+        """
+        if is_request_from_mobile_app(request):
+            user_agent = request.META.get('HTTP_USER_AGENT')
+            platform, user_app_version = self.form_factor(user_agent)
+
+            # if user app version does not match to X.X.X.Y (acceptable format) then let pass the request
+            # as it might belong to older apps
+            user_app_version = self.validate_version(user_app_version)
+            if user_app_version:
+                user_app_version_tuple = self.version_tuple(user_app_version)
+                app_version_config = MobileAppVersionConfig.current()
+
+                min_supported_version = app_version_config.get_min_supported_version(platform)
+                if user_app_version_tuple < self.version_tuple(min_supported_version):
+                    return HttpResponse(status=426)
+
+                next_supported_version = app_version_config.get_next_supported_version(platform)
+                next_update_required = app_version_config.get_next_update_required(platform).replace(tzinfo=None)
+                if user_app_version_tuple < self.version_tuple(next_supported_version):
+                    if datetime.now() > next_update_required:
+                        return HttpResponse(status=426)
+
+    def process_response(self, _request, response):
+        """
+        If request is from mobile native app and confront to X.X.X.Y version format (where X is a number and
+        Y is optional alphanumeric), then add headers to response;
+        1. EDX-APP-LATEST-VERSION; if user app version < latest available version
+        2. EDX-APP-UPGRADE-DATE; if user app version < next supported version
+        """
+        if is_request_from_mobile_app(_request):
+            user_agent = _request.META.get('HTTP_USER_AGENT')
+            platform, user_app_version = self.form_factor(user_agent)
+            # if user app version does not match to X.X.X.Y (acceptable format) then do not add version info to
+            # headers as this request might belong to older apps
+            user_app_version = self.validate_version(user_app_version)
+            if user_app_version:
+                app_version_config = MobileAppVersionConfig.current()
+                latest_version = app_version_config.get_latest_version(platform)
+                user_app_version_tuple = self.version_tuple(user_app_version)
+                if user_app_version_tuple < self.version_tuple(latest_version):
+                    response["EDX-APP-LATEST-VERSION"] = latest_version
+                next_supported_version = app_version_config.get_next_supported_version(platform)
+                if user_app_version_tuple <= self.version_tuple(next_supported_version):
+                    next_update_required = app_version_config.get_next_update_required(platform).replace(tzinfo=None)
```

Comment:
Updating the timezone should happen inside the config model method so that it does not need to be repeated.